src/gatsby/hooks/stars.ts

```diff
@@ -0,0 +1,36 @@
+import { useEffect, useState } from 'react'
+import { useStaticQuery, graphql } from 'gatsby'
+import fetch from 'isomorphic-fetch'
+
+export default function useStars(): number {
+  // Get the amount of stars from build time
+  const staticStars = useStaticQuery(graphql`
+    query GithubStarsQuery {
+      githubData {
+        data {
+          repository {
+            stargazers {
+              totalCount
+            }
+          }
+        }
+      }
+    }
+  `).githubData.data.repository.stargazers.totalCount
+
+  // Maintain an updatable state so we can update stars on delivery
+  const [stars, setStars] = useState(staticStars)
+
+  // Run an IIFE to update from the server on the client side.
+  useEffect(() => {
+    ;(async (): Promise<void> => {
+      const res = await fetch(`/api/github/stars?current=${stars}`)
```

Comment:
since we include query - need to make sure that cache is handled properly

Reply:
figured this out ... but I think we don't need this at the moment? we save a few bytes but complicate code

Reply:
I can remove the cache checking, if that's what you're talking about, but this IIFE in general is what fetches the data that updates stars if we got more since last build.

Reply:
yep, I mean the `?current=${stars}` part!